core/src/main/java/jenkins/slaves/JnlpSlaveAgentProtocol4.java

```diff
@@ -175,11 +149,18 @@ public String getDisplayName() {
 
     @Override
     public String getName() {
-        return handler.getName();
+        return "JNLP4-connect"; // matches JnlpProtocol4Handler.getName
     }
 
     @Override
     public void handle(Socket socket) throws IOException, InterruptedException {
+        try {
+            init();
+        } catch (IOException x) {
+            throw x;
+        } catch (Exception x) {
```

Comment:
`KeyStoreException, KeyManagementException` but also `IllegalStateException`…just seemed simpler to catch anything.